internal/provider/incident_alert_route_resource_test.go

```diff
@@ -211,6 +211,206 @@ func TestAccIncidentAlertRouteResourceConditionalConfig(t *testing.T) {
+					resource.TestCheckResourceAttrSet("incident_custom_field.alpha_field2", "id"),
+
+					// Check that we have exactly 2 custom fields
+					resource.TestCheckResourceAttr("incident_alert_route.custom_fields_alpha_test", "incident_template.custom_fields.#", "2"),
+				),
+			},
+			// Second step - refresh state and verify no diff
+			{
+				RefreshState: true,
+				PlanOnly:     true,
+				RefreshPlanChecks: resource.RefreshPlanChecks{
+					PostRefresh: []plancheck.PlanCheck{
+						plancheck.ExpectEmptyPlan(),
+					},
+				},
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Only check custom fields, not other attributes
+					resource.TestCheckResourceAttr("incident_alert_route.custom_fields_alpha_test", "incident_template.custom_fields.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccIncidentAlertRouteWithAlphabeticalCustomFields(name string) string {
+	return `                                                                                                                                                                           
+  resource "incident_custom_field" "alpha_field1" {                                                                                                                                    
+    name        = "Alpha Test Field 1"                                                                                                                                                 
+    description = "First alphabetical custom field"                                                                                                                                    
+    field_type  = "text"                                                                                                                                                               
+  }                                                                                                                                                                                    
+                                                                                                                                                                                       
+  resource "incident_custom_field" "alpha_field2" {                                                                                                                                    
+    name        = "Alpha Test Field 2"                                                                                                                                                 
+    description = "Second alphabetical custom field"                                                                                                                                   
+    field_type  = "text"                                                                                                                                                               
```

Comment:
```suggestion
    field_type  = "text"
    depends_on = [incident_custom_field.alpha_field_1]
```

This'll ensure field 2 is always created after field 1